src/javascript/app_2/App/Components/Routes/__tests__/helpers.spec.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { expect }      from 'chai';
+import * as Helpers    from '../helpers';
+import getRoutesConfig from '../../../Constants/routes_config';
+import Trade           from 'Modules/Trading';
+
+describe('Helpers', () => {
+    describe('normalizePath', () => {
+        it('should return / as if path is empty', () => {
+            expect(Helpers.normalizePath('')).to.equal('/');
+        });
+        it('should return / + path as if path does not have /', () => {
+            expect(Helpers.normalizePath('trade')).to.equal('/trade');
+        });
+        it('should return / + path as if path does have /', () => {
+            expect(Helpers.normalizePath('/trade')).to.equal('/trade');
+        });
+    });
+
+    describe('findRouteByPath', () => {
+        it('should return undefined when path is not in routes_config', () => {
+            expect(Helpers.findRouteByPath('invalidRoute', getRoutesConfig())).to.be.undefined;
+        });
+        it('should return route_info when path is in routes_config and is not nested', () => {
+            expect(Helpers.findRouteByPath('/trade', getRoutesConfig())).to.eql({
+                path: '/trade',
+                component: Trade,
+                title: 'Trade',
+                exact: true
+            });
+        });
+        it('should return route_info of parent route when path is in routes_config child level and is nested', () => {
+            expect(Helpers.findRouteByPath('/settings/personal', getRoutesConfig())).to.have.all.keys('path', 'component', 'is_authenticated', 'routes');
+            expect(Helpers.findRouteByPath('/settings/personal', getRoutesConfig()).routes).to.be.instanceof(Array);
+            expect(Helpers.findRouteByPath('/settings/personal', getRoutesConfig()).routes).to.have.length(9);
```

Comment:
@nazaninreihani 
It should be better to prevent from hardcoding data (e.g. length of sub-routes in this test) that will be changed probably in future. This kind of hardcoded data causes tests to fail without any reasons.